src/protocol/sort/generate_permutation.rs

```diff
@@ -98,21 +98,22 @@ pub async fn generate_permutation<F: Field>(
     num_bits: u32,
 ) -> Result<Vec<Replicated<F>>, Error> {
     let ctx_0 = ctx.narrow(&Sort(0));
-    let bit_0 =
-        convert_shares_for_a_bit(ctx_0.narrow(&ModulusConversion), input, num_bits, 0).await?;
+    let triple_0 = convert_bits_local(ctx_0.role(), 0, input);
+    let bit_0 = convert_bits(ctx_0.narrow(&ModulusConversion), &triple_0).await?;
     let bit_0_permutation = bit_permutation(ctx_0.narrow(&BitPermutationStep), &bit_0).await?;
     let input_len = u32::try_from(input.len()).unwrap(); // safe, we don't sort more that 1B rows
 
     let mut composed_less_significant_bits_permutation = bit_0_permutation;
     for bit_num in 1..num_bits {
         let ctx_bit = ctx.narrow(&Sort(bit_num));
+        let triples = convert_bits_local(ctx.role(), bit_num, input);
```

Comment:
I think the name `input` is too generic and unhelpful. How about `keys`? Or `keys_to_sort_by`